src/makeCalculator.js

```diff
@@ -4,7 +4,45 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(number) {
+      this.result -= number;
+    },
+
+    multiply(number) {
+      this.result *= number;
+    },
+
+    divide(number) {
+      if (number === 0) {
+        throw new Error('Cannot divide by zero');
+      }
+
+      this.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      if (typeof callback === 'function') {
+        callback.call(this, number);
+      }
+
+      return this;
```

Comment:
Use the method shorthand syntax for defining methods. Instead of `operate: function(callback, number) { ... }`, you can simply use `operate(callback, number) { ... }`. The check for the callback being a function is correct and chaining is properly implemented by returning `this`.